compat/sched_affinity.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_COMPAT_SCHED_AFFINITY_H
+#define SQUID_COMPAT_SCHED_AFFINITY_H
+
+#if !HAVE_SCHED_SETAFFINITY
+
+#include <stdbool.h>
+#if HAVE_SYS_CPUSET_H
+#include <sys/param.h>
+#include <sys/cpuset.h>
+#undef HAVE_CPU_SET_T
+#define HAVE_CPU_SET_T 1
+#undef HAVE_CPU_AFFINITY
+#define HAVE_CPU_AFFINITY 1
+#define cpu_set_t cpuset_t
+#else
+typedef struct {
+    int bits;
+} cpu_set_t;
+#endif
```

Comment:
The struct definition was previously wrapped in a type-specific check for cpu_set_t existence, which is better portability practice than relying on a header name.

Please separate this from the headers include block and return it to being as-needed. Doing so will fix the issues around HAVE_CPU_SET_T as you can see:
```suggestion

#if !HAVE_CPU_SET_T
// Linux 2.6 and newer renamed it cpuset_t
#if HAVE_CPUSET_T
typedef cpuset_t cpu_set_t;
#else
// No alternative known, define it ourselves
typedef struct {
    int bits;
} cpu_set_t;
#endif
#define HAVE_CPU_SET_T 1
#endif
```